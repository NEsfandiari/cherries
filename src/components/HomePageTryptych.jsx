import React from 'react'
import styled from 'styled-components'
import { StaticQuery, graphql } from 'gatsby'

import { TryptychPanel } from './molecules'
import { HomePageButton } from './atoms'

const Container = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
  align-items: center;
  background-color: #ffe2e2;
  width: 100vw;
  padding: 3rem;
  height: 35rem;
  #tryptych-header {
    margin: 0;
    color: #e20031;
    font-size: 3rem;
    width: 100%;
    text-align: center;
    font-style: italic;
  }
  #tryptych-panel-container {
    display: flex;
    justify-content: space-evenly;
    width: 70%;
  }
  .tryptych-panel-text {
    line-height: 1.5em;
    color: #47525e;
  }
`

export default () => (
  <StaticQuery
    query={graphql`
      {
        contentfulHomePage(pageTitle: { eq: "Home Page" }) {
          tryptychIcons {
            fluid {
              src
            }
          }
          tryptych {
            content {
              nodeType
              content {
                value
              }
            }
          }
        }
      }
    `}
    render={data => {
      const tryptychContent = data.contentfulHomePage.tryptych.content
      let tryptychHeader
      let tryptychIcons = data.contentfulHomePage.tryptychIcons
      let tryptychCopy = []

      // Seperate content into its seperate nodetypes
      tryptychContent.forEach(item => {
        if (item.nodeType === 'heading-4')
          tryptychHeader = item.content[0].value
        else if (item.nodeType === 'paragraph')
          tryptychCopy.push(item.content[0].value)
      })

      let tryptychPanels = []
      tryptychIcons.forEach((icon, i) => {
        tryptychPanels.push(
          <TryptychPanel
            key={i}
            imageUrl={icon.fluid.src}
            imageName={icon.fileName}
            text={tryptychCopy[i]}
          />
        )
      })
      return (
        <Container>
          <h2 id="tryptych-header">{tryptychHeader}</h2>
          <div id="tryptych-panel-container">{tryptychPanels}</div>
          <HomePageButton />
        </Container>
      )
    }}
  />
)
