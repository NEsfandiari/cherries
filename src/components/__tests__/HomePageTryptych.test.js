import React from 'react'
import renderer from 'react-test-renderer'
import { HomePageTryptych, PureHomePageTryptych } from '../HomePageTryptych'

describe('HomePageHero', () => {
  it('renders correctly', () => {
    const data = {
      contentfulHomePage: {
        tryptych: {
          content: [
            {
              nodeType: 'heading-4',
              data: {
                target: null,
              },
              content: [
                {
                  value: 'Jewelry, but make it fun.',
                },
              ],
            },
            {
              nodeType: 'embedded-asset-block',
              data: {
                target: {
                  fields: {
                    file: {
                      en_US: {
                        url:
                          '//images.ctfassets.net/4yejk6rho2rp/29LnpcJWdua0WY4Ugm4Igs/0c47d026abad1822dca7bc2a8fb870cf/sparkle-emoji.png',
                        fileName: 'sparkle-emoji.png',
                      },
                    },
                  },
                },
              },
              content: [],
            },
            {
              nodeType: 'paragraph',
              data: {
                target: null,
              },
              content: [
                {
                  value:
                    "We make cute, unique jewelry for every occasionâ€”that good ish you can't get anywhere else.",
                },
              ],
            },
            {
              nodeType: 'embedded-asset-block',
              data: {
                target: {
                  fields: {
                    file: {
                      en_US: {
                        url:
                          '//images.ctfassets.net/4yejk6rho2rp/2LVH1dUby8UawuqSQGO0Yi/15aa6e94241b664921024c13add2907e/envelope-emoji.png',
                        fileName: 'envelope-emoji.png',
                      },
                    },
                  },
                },
              },
              content: [],
            },
            {
              nodeType: 'paragraph',
              data: {
                target: null,
              },
              content: [
                {
                  value:
                    'Subscribe to monthly deliveries to save that $$$ and keep your looks fresh.',
                },
              ],
            },
            {
              nodeType: 'embedded-asset-block',
              data: {
                target: {
                  fields: {
                    file: {
                      en_US: {
                        url:
                          '//images.ctfassets.net/4yejk6rho2rp/2Uv99J8a9iEeySUusc8UcW/2a4d76f25e3f9e02055406cacb344ca0/delivery-truck-emoji.png',
                        fileName: 'delivery-truck-emoji.png',
                      },
                    },
                  },
                },
              },
              content: [],
            },
            {
              nodeType: 'paragraph',
              data: {
                target: null,
              },
              content: [
                {
                  value:
                    'We offer free shipping to subscribers on all US orders, because we got your back girl.',
                },
              ],
            },
          ],
        },
      },
    }

    const tree = renderer.create(<PureHomePageTryptych data={data} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
