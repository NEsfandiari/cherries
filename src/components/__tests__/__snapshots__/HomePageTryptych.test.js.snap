// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePageHero renders correctly 1`] = `
<div
  className="sc-hSdWYo dhuhYa"
>
  <h2
    id="tryptych-header"
  >
    Jewelry, but make it fun.
  </h2>
  <div
    id="tryptych-panel-container"
  >
    <div
      className="sc-bdVaJa ftRuNj"
    >
      <div
        className="tryptych-panel"
      >
        <img
          alt="sparkle-emoji.png"
          className="tryptych-panel-image"
          src="//images.ctfassets.net/4yejk6rho2rp/29LnpcJWdua0WY4Ugm4Igs/0c47d026abad1822dca7bc2a8fb870cf/sparkle-emoji.png"
        />
        <p
          className="tryptych-panel-text"
        >
          We make cute, unique jewelry for every occasionâ€”that good ish you can't get anywhere else.
        </p>
      </div>
    </div>
    <div
      className="sc-bdVaJa ftRuNj"
    >
      <div
        className="tryptych-panel"
      >
        <img
          alt="envelope-emoji.png"
          className="tryptych-panel-image"
          src="//images.ctfassets.net/4yejk6rho2rp/2LVH1dUby8UawuqSQGO0Yi/15aa6e94241b664921024c13add2907e/envelope-emoji.png"
        />
        <p
          className="tryptych-panel-text"
        >
          Subscribe to monthly deliveries to save that $$$ and keep your looks fresh.
        </p>
      </div>
    </div>
    <div
      className="sc-bdVaJa ftRuNj"
    >
      <div
        className="tryptych-panel"
      >
        <img
          alt="delivery-truck-emoji.png"
          className="tryptych-panel-image"
          src="//images.ctfassets.net/4yejk6rho2rp/2Uv99J8a9iEeySUusc8UcW/2a4d76f25e3f9e02055406cacb344ca0/delivery-truck-emoji.png"
        />
        <p
          className="tryptych-panel-text"
        >
          We offer free shipping to subscribers on all US orders, because we got your back girl.
        </p>
      </div>
    </div>
  </div>
  <div
    className="sc-bZQynM cjDaME"
  >
    <a
      href="/subscribe"
    >
      GET STARTED
    </a>
  </div>
</div>
`;
